### PR TITLE
LUCENE-9278: Make javadoc folder structure follow Gradle project path

### PR DESCRIPTION
**Pull request by mocobeta**
# Description

See https://issues.apache.org/jira/browse/LUCENE-9278

### CODE REVIEW

File: gradle/render-javadoc.gradle [hunk 1]
```diff
@@ -17,11 +17,15 @@
 
 // generate javadocs by using Ant javadoc task
 
+// utility function to convert project path to document output dir
+// e.g.: ':lucene:analysis:common' => 'analysis/common'
+def pathToDocdir = { path -> path.split(':').drop(2).join('/') }
```
Comment (mocobeta) [inline gradle/render-javadoc.gradle]: There might be better way?

File: gradle/render-javadoc.gradle [hunk 2]
```diff
@@ -294,35 +305,41 @@ configure(project(':solr:core')) {
...
     }
   }
 }
 
+
 configure(project(':solr:contrib:extraction')) {
   plugins.withType(JavaPlugin) {
     ext {
-      javadocDestDir = "${javadocRoot}/solr-cell"
+      javadocDestDir = "${javadocRoot}/contrib/cell"
```
Comment (mocobeta) [inline gradle/render-javadoc.gradle]: I'm not sure this tweak is needed for gradle build. Simply, "solr/contrib/extraction" (same as other contrib modules) may be okay?

File: gradle/render-javadoc.gradle (hunk 1 above)
Comment (dweiss) [reply gradle/render-javadoc.gradle]: This seems fine to me.

File: gradle/render-javadoc.gradle (hunk 2 above)
Comment (dweiss) [reply gradle/render-javadoc.gradle]: The "solr-cell" is what it's called in ant. I'm not sure why. I would prefer to stick to folder/ project path naming here too (so contrib/extraction for me).

File: gradle/render-javadoc.gradle [hunk 3]
```diff
@@ -17,11 +17,15 @@
...
+// utility function to convert project path to document output dir
+// e.g.: ':lucene:analysis:common' => 'analysis/common'
+def pathToDocdir = { path -> path.split(':').drop(2).join('/') }
+
 allprojects {
   plugins.withType(JavaPlugin) {
     ext {
       javadocRoot = project.path.startsWith(':lucene') ? project(':lucene').file("build/docs") : project(':solr').file("build/docs")
-      javadocDestDir = "${javadocRoot}/${project.name}"
+      javadocDestDir = "${javadocRoot}/" + pathToDocdir(project.path)
```
Comment (dweiss) [inline gradle/render-javadoc.gradle]: You can call the function inside the evaluated string too; "${javadocRoot}/${pathToDocDir(project.path)}"

File: gradle/render-javadoc.gradle (hunk 2 above)
Comment (mocobeta) [reply gradle/render-javadoc.gradle]: I removed the entire configure block, so now the docs will go to `contrib/extraction`.

File: gradle/render-javadoc.gradle (hunk 3 above)
Comment (mocobeta) [reply gradle/render-javadoc.gradle]: Fixed, thanks.

**Comment by mocobeta** [comment]
I mean to close this PR because we will eventually replace all relative paths here with absolute URLs, sooner or later. I'll try to make some changes to renderJavadoc task to drop all relative paths (and move all generated docs under _project_/build again).

**Pull request closed without merging**